### PR TITLE
Override log levels after log handler is created

### PR DESCRIPTION
**Pull request by Abacn**
Follow up of #18222

* This fixes error message about log level overrides not shown when
  executed with Dataflow runner

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @TheNeuralBit

**Comment by TheNeuralBit** [comment]
Would it be possible to add a test that fails without this change?

**Comment by Abacn** [comment]
retest this please

**Comment by Abacn** [comment]
> Would it be possible to add a test that fails without this change?

Addressed. please take another look, thanks!

File: sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py [hunk 1]
```diff
@@ -100,6 +101,30 @@ def test_runtime_values(self):
     self.assertTrue(test_runtime_provider.is_accessible())
     self.assertEqual(test_runtime_provider.get(), 37)
 
+  def test_crerate_sdk_harness_log_handler_received_log(self):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/worker/sdk_worker_main_test.py:104]:
```suggestion
  def test_create_sdk_harness_log_handler_received_log(self):
```

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**